source/MaterialXShaderGen/ShaderGenerators/Glsl/GeomAttrValueGlsl.cpp

```diff
@@ -0,0 +1,60 @@
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/GeomAttrValueGlsl.h>
+
+namespace MaterialX
+{
+
+SgImplementationPtr GeomAttrValueGlsl::creator()
+{
+    return std::make_shared<GeomAttrValueGlsl>();
+}
+
+void GeomAttrValueGlsl::createVariables(const SgNode& node, ShaderGenerator& shadergen, Shader& shader_)
+{
+    HwShader& shader = static_cast<HwShader&>(shader_);
+
+    const SgInput* attrNameInput = node.getInput(ATTRNAME);
+    if (!attrNameInput)
+    {
+        throw ExceptionShaderGenError("No 'attrname' parameter found on geomattrvalue node '" + node.getName() + "', don't know what attribute to bind");
+    }
+    const string name = attrNameInput->value->getValueString();
+    const string type = shadergen.getSyntax()->getTypeName(node.getOutput()->type);
+    const string variable = "geomattr_" + name;
+
+    shader.createAppData(type, "i_" + variable);
+    shader.createVertexData(type, variable);
```

Comment:
Just checking here since I was never totally clear on geomattr. We are assuming these are always vertex streams and not single uniforms as the spec makes them sound more like single uniforms.

Reply:
That is a good question. I was under the impression that the OSL reference implementation treated these as varyings. But now when I looked at it again, and also re-read that section in the Mtlx specification, it looks like these should actually be constants. So I will change that.

It also means that where are some work needed on application side to support this, because we need to extract the values to bind from the MaterialX document (since these values can be different per geometry being rendered). But that's another task.